common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -822,14 +815,18 @@ function (HTML5Video, Resizer) {
 
     function updatePlayTime(time, skip_seek) {
         var videoPlayer = this.videoPlayer,
-            duration = this.videoPlayer.duration(),
+            endTime = this.videoPlayer.duration(),
             youTubeId;
 
+        if (this.config.endTime) {
+            endTime = Math.min(this.config.endTime, endTime);
```

Comment:
It's possible, but I've found it to be a bit difficult as the logic is repeated across several video player modules (`03_video_player.js`, `06_video_progress_slider.js`). In order to do so, we would have to attach the said utility function to `state.videoPlayer` in the file `03_video_player.js`, and this doesn't seem to be the right place to put a dependency there. I'm open to it if it seems like a necessary unification.